MSAL/src/instance/MSALAccountsProvider.m

```diff
@@ -148,13 +164,44 @@ - (MSALAccount *)accountForUsername:(NSString *)username
     
     if ([msidAccounts count])
     {
-        MSALAccount *msalAccount = [[MSALAccount alloc] initWithMSIDAccount:msidAccounts[0]];
-        return msalAccount;
+        NSArray<MSALAccount *> *msalAccounts = [self msalAccountsFromMSIDAccounts:msidAccounts];
```

Comment:
`accountForHomeAccountId:` & `accountForUsername:` have similar logic, should we refactor them to reduce copy/paste?

Reply:
:+1: addressed :)